### DISCUSSION TITLE
Possible multithreading problem

### DISCUSSION THREAD

**Discussion by artheru** [sebastienros/jint#1177] [Q&A] [answered]
I'm using jint as a script interpretor for robot movement arrangement, it comes into this issue:
![09e2d3c01213b35bf1d6696796bb0a2](https://user-images.githubusercontent.com/6483114/169768255-955bb868-8f82-457b-99f2-627dce9a6cae.png)

I guess there could be some multi-thread related problem here. Method agv.Queue(a,b,c,d....) could run a,b,c,d simultaneously.

**Comment by lahma** [comment, marked as answer]
Jint is not thread safe, and it's by design, so you need to ensure no multiple threads are accessing the engine at the same time.
